### DISCUSSION TITLE
Questions about how to use veramo cli

### DISCUSSION THREAD

**Discussion by cbw1030** [decentralized-identity/veramo#657] [Q&A] [answered]
Hello, I'm currently running an agent with veramo cli.
I have some questions.

1. The link below tells us to write agent scheme to connect agent.
What is the meaning of agent system and where can I find it?
https://explore.veramo.dev/connect

2. How can I make a personal wallet in veramo cli?

3. You can create a public key through the 'veramo did add-key' command, what is the meaning of this public key and how can I get a private key that matches that key?

Thanks for taking time out from your busy schedule.

**Comment by mirceanis** [comment, marked as answer]
Hello, these are all excellent questions.

1. By default, the schema is served at `/open-api.json`. This is one of the routes that is served when you run `veramo server`.
You can see all the default routes in the [`agent.yml / server`](https://github.com/uport-project/veramo/blob/232de6656cf14c195d478915c8ccb1b203677483/packages/cli/default/default.yml#L75) config section.
The default routes are `/messaging`, `/agent`, `/open-api.json`, `/api-docs`

Normally, when you just run `veramo server` the agent uses `http://localhost:3332` to listen for requests. You can browse to `http://localhost:3332/api-docs` to see it in action.
If you want to use `https://explore.veramo.dev` with your agent, your agent needs to be accessible from the web.
We've found it easy to work with [ngrok.io](https://ngrok.io/) to get a public endpoint for a locally running agent.
Alternatively, you can try to [deploy the agent to a cloud](https://github.com/uport-project/veramo-agent-deploy)

2. That depends a lot on what your personal wallet needs to do. By default a `@veramo/cli` agent is able to create DIDs and issue, receive and present credentials related to these DIDs. Some extra configuration steps may be necessary to get full functionality, like my answer to question 1.

3. The `veramo did add-key` command is supposed to create a private key and add the corresponding public key to the DID document of the DID you are managing. Depending on the type of key you add, this can later be used for signing or [de]encryption.
It is not recommended to try to use the private key directly.
This key(or one of the other managed keys) is used when your agent issues credentials (for example, `veramo credential create`), presentations, during selective disclosure, or for communication using DIDComm.

I hope this info helps. Perhaps if you describe your use case in more detail I can provide more concrete answers.

**Reply by cbw1030** [reply]
Thank you for your answer.
Run with 'veramo server' command and verify good connection with 'localhost:3332/api-docs'.
We also checked the source code for 'localhost:3332/open-api.json'.
But there are still some things I don't understand, so I'm leaving a comment.

1. Attempting to connect to 'localhost:3332/messaging' results in 'Cannot GET/messaging' error. Why is that?
Attempting to connect to 'localhost:3332/agent' results in an 'Unauthorized' error.
I ran it on the local server, but why does the above error occur when I connect with '/agent'?

2. When you access '/api-docs', the 'Authorize' button exists.
I'd like to know what the button means. It's my first time using Swagger UI.
If you set the value and refresh it, it will be released.

3. As you answered, I'm looking for schema in /open-api.json, but I'm not sure what it means.
Can you give me an example, assuming you enter schema in 'https://expore.veramo.dev/connect'?

4. Enter the 'npx agent-expore service' command to access localhost:5000.
I wonder the difference between the localhost:5000 site and the https://expore.veramo.dev/connect site.

5. As you have answered, you can access as follows at node.js by referring to ngrok.io.
localhost:3332/api-docs -> https://15d73b196ada.ngrok.io/api-docs/
But I don't know what to do afterwards.
Can I communicate with my local DID, Credential and https://expore.veramo.dev/connect here?

Thank you for reading the long question.

**Reply by mirceanis** [reply]
Hi, I answered in your other post #663
Please avoid posting the same question multiple times as it makes it harder to answer.
